amethyst_core/src/transform/components/local_transform.rs

```diff
@@ -148,33 +245,11 @@ impl Transform {
     }
 
     /// Set the rotation using Euler x, y, z.
-    pub fn set_rotation<D: Into<Deg<f32>>>(&mut self, x: D, y: D, z: D) -> &mut Self {
-        let rotation = Quaternion::from_angle_x(x.into()) * Quaternion::from_angle_y(y.into())
-            * Quaternion::from_angle_z(z.into());
-
-        self.rotation = rotation.into();
+    pub fn set_rotation<A: Into<Rad<f32>>>(&mut self, x: A, y: A, z: A) -> &mut Self {
```

Comment:
Should we take `cgmath::Euler` instead to type enforce that it should be Euler angles?

Reply:
I don't know what the correct answer is here. Do you still get the 'expected' result if you use angles from outside the expected ranges?

Reply:
The problem with the current solution is that it suffers from gimbal lock, because the rotations are applied in sequence. Using the Euler conversion formula in cgmath is likely to not suffer from this problem. See https://docs.rs/cgmath/0.16.1/src/cgmath/quaternion.rs.html#280.

Reply:
You've convinced me :)